version: '3.1'

services:

   db:
    image: mysql:5.7
    ports:
      - '3306:3306'
    env_file: .env
    restart: always
    volumes:
      - ./db_django:/var/lib/mysql
      - ./config/mysql/custom.cnf:/etc/mysql/conf.d/custom.cnf
   
   web:
      image: python:3.9.6
      env_file: .env
      volumes:
         - ./:/opt/google_calendar_sync
         - ./tmp:/tmp
      working_dir: /opt/google_calendar_sync
      expose:
         - 8000
      container_name: google_calendar_sync-web
      command: /bin/bash ./scripts/run_django.sh
      depends_on:
      - db

   redis:
      image: redis:latest
      container_name: google_calendar_sync-redis
      env_file: .env

   worker_sync:
      env_file: .env
      volumes:
         - ./:/opt/google_calendar_sync
         - ./tmp:/tmp
      image: python:3.9.6
      working_dir: /opt/google_calendar_sync
      container_name: google_calendar_sync-worker-sync
      command: /bin/bash ./scripts/run_celery.sh sync
      depends_on:
         - redis

   worker_limited:
      env_file: .env
      volumes:
         - ./:/opt/google_calendar_sync
         - ./tmp:/tmp
      image: python:3.9.6
      working_dir: /opt/google_calendar_sync
      container_name: google_calendar_sync-worker-limited
      command: /bin/bash ./scripts/run_celery.sh limited
      depends_on:
         - redis

   beat:
      env_file: .env
      volumes:
         - ./:/opt/google_calendar_sync
         - ./tmp:/tmp
      image: python:3.9.6
      working_dir: /opt/google_calendar_sync
      container_name: google_calendar_sync-beat
      command: /bin/bash ./scripts/run_celery.sh beat
      depends_on:
         - redis

   nginx:
      image: nginx:latest
      volumes:
         - ./config/nginx/production:/etc/nginx/conf.d
         - ./config/nginx/nginx.conf:/etc/nginx/nginx.conf
         - ./:/opt/google_calendar_sync
      ports:
         - "0.0.0.0:80:80"
      container_name: google_calendar_sync-nginx
      depends_on:
         - web
      restart: "on-failure"